### ISSUE TITLE
Train FlairEmbedding with document boundaries separator token

### ISSUE THREAD

**Issue by otahmasebi** [flairNLP/flair#1538]
Hello
I have tried to train new Flair Embeddings and as mentioned In https://github.com/flairNLP/flair/blob/master/resources/docs/TUTORIAL_9_TRAINING_LM_EMBEDDINGS.md , it is possible to use a document separator token like [SEP]. I use this token in my corpus to separate document boundaries, how should we introduce this token to LanguageModelTrainer ?!
Is there any docs to share more details to how sentences  feed to model in this way when shuffle_lines variabble is true?!

**Comment by alanakbik** [comment]
Hello @otahmasebi good points, thanks for noticing! Setting `shuffle_lines` to `True` simply shuffles a file by newlines and does not respect document boundaries. I think we need to change this.

**Comment by otahmasebi** [comment]
thanks for your reply. @alanakbik

**Comment by alanakbik** [comment]
Just merged a PR that adds the option to set the delimiter, like this: 

```python
# your document delimiter
delimiter = '[SEP]'

# set it when you load the corpus
corpus = TextCorpus(
    "data/corpora/conala-corpus/",
    dictionary,
    is_forward_lm,
    character_level=True,
    document_delimiter=delimiter,
)

# set it when you init the language model
language_model = LanguageModel(
    dictionary,
    is_forward_lm=True,
    hidden_size=512,
    nlayers=1,
    document_delimiter=delimiter
)

# train your language model as always
trainer = LanguageModelTrainer(language_model, corpus)
```

This is now available in master branch.

**Comment by UrszulaCzerwinska** [comment]
Hi, I am working with long documents: learning LM and NER. I tried to add document separator for LM and for TextCorpus. 
What I observe is that instead of having sentences loaded to flair (that are separated in my IOB by \n), I have whole documents seen as a sentence. 

Having a document seen as a sentence, for long documents seems less intuitive. I didn't inspect the impact on the score yet for the Sequence Tagger.

From a quick test in flair character LM training, it runs faster. However, it just truncates more randomly the text chunks that are now whole documents instead of sentences (so longer that max length most of the time) and the score is poorer. 

Am I wrongly interpeting the use of doument delimiter? Is it not suited for long documents? 

I was imaging a sort of added hierarhical layers: corpus => document => sentence => token and it seems the document delimiter just allows to replace \n separator by a custom one, is that right?

**Comment by alanakbik** [comment]
Hello @UrszulaCzerwinska sorry for the late reply. If you train the character LM, the whole corpus is used as one long sequence of characters, so there is no explicit hierarchy like you describe and also no truncation based on the document delimiter. The delimiter is only used internally when generating embeddings as a signal that a document is starting. Meaning: If your text corpus you use to train contains an explicit document delimiter, you can provide it to improve the embeddings. Otherwise, a newline is always a good fallback option. 

When you read NER, the LM delimiter has no impact since it is only used internally by the language model. Your NER files should be formatted regularly.
